### ISSUE TITLE
mod_wsgi-express complains about undefined symbol:  PyFrame_GetLineNumber

### ISSUE THREAD

**Issue by user7766666** [GrahamDumpleton/mod_wsgi#275]
Using RedHat Enterprise Linux Server 6,9 (Santiago) and Apache version:

$ httpd -v
Server version: Apache/2.2.15 (Unix)
Server built:   Sep 19 2017 11:18:5

After installing mod_wsig-4.6.2 sucessfully

$ pip install mod_wsgi
You are using pip version 7.1.0, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting mod-wsgi
/usr/lib/python2.6/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading mod_wsgi-4.6.2.tar.gz (2.6MB)
    100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 2.6MB 219kB/s 
Installing collected packages: mod-wsgi
  Running setup.py install for mod-wsgi
Successfully installed mod-wsgi-4.6.2

 and trying to start it with the mod_wsgi_express script, I get the following fault:

mod_wsgi-express start-server
Server URL         : http://localhost:8000/
Server Root        : /tmp/mod_wsgi-localhost:8000:299
Server Conf        : /tmp/mod_wsgi-localhost:8000:299/httpd.conf
Error Log File     : /tmp/mod_wsgi-localhost:8000:299/error_log (warn)
Request Capacity   : 5 (1 process * 5 threads)
Request Timeout    : 60 (seconds)
Startup Timeout    : 15 (seconds)
Queue Backlog      : 100 (connections)
Queue Timeout      : 45 (seconds)
Server Capacity    : 20 (event/worker), 20 (prefork)
Server Backlog     : 500 (connections)
Locale Setting     : en_US.UTF8
httpd (mod_wsgi-express) : Syntax error on line 159 of /tmp/mod_wsgi-localhost:8000:299/httpd.conf: Cannot load /usr/lib64/python2.6/site-packages/mod_wsgi/server/mod_wsgi-py26.so into server: /usr/lib64/python2.6/site-packages/mod_wsgi/server/mod_wsgi-py26.so: undefined symbol: PyFrame_GetLineNumber

It seems like it has problems loading the  Python library, but I can't figure out why exactly. Any help appreciated.

**Comment by GrahamDumpleton** [comment]
It is because you are using Python 2.6, which is getting more than a bit old now. A change was made to use ``PyFrame_GetLineNumber()`` to calculate line number of Python error from stack frames. I understood it existed back to older versions than Python 2.6, and the Python code said to use it rather than access line number of frame object direct, which could be wrong.

For now, pin your mod_wsgi version to:

```
mod_wsgi<4.6.0
```

**Comment by user7766666** [comment]
Ok, I'll try downloading mod_wsgi 4.5.24

**Comment by GrahamDumpleton** [comment]
I have fixed in ``develop`` branch. Too tired to do a release tonight. If really wanted to try is, use:

* https://github.com/GrahamDumpleton/mod_wsgi/archive/develop.zip

**Comment by user7766666** [comment]
version 4.5.24 was installed correctly.

Now I can run mod_wsgi-express start-server and it seems to be working. But my next step is maybe not related to mod_wsgi.

My next issue is trying to figure out how to get my little Flask/Python application to work, I don't seem to get any errors or any messages at all to give me any hints about what is going on. Is there a way to figure out if Apache is running/calling mod_wsgi?

My configuration files look like this:

**Comment by user7766666** [comment]
Forget my last line from above

**Comment by GrahamDumpleton** [comment]
Are you trying to use ``mod_wsgi-express start-server``, or trying to integrate mod_wsgi and your Flask application into your system Apache?

When running ``mod_wsgi-express start-server`` use the ``--log-to-terminal`` line to have it show the error log to the terminal. Also add ``--log-level info`` and mod_wsgi will log more about what it is doing such as when loading the WSGI script file, when restarting daemon processes. To track requests, can also add ``--access-log`` option.

Beyond those hints, you would need to tell me what command line you ran if using ``mod_wsgi-express start-server``. What error you get in the browser, and what error in the logs.

**Comment by user7766666** [comment]
I just run mod_wsgi-express start-server to see that mod_wsgi was working, which it is.

I have developed a small application using Flask, which is working fine when  executing"flask run", showing a very simple HTML page. My next step is to have it integrated to myownpage.com/flasktest which is hosted by Apache. My question is how to verify that Apache is calling my flask app using mod_wsgi. Because when I try to access myownpage.com/flasktest nothing happens, it just shows a directtory listing and not the flask HTML page.

Another issue that bothers me is the following after running mod_wsig_express install-module:

LoadModule wsgi_module "/usr/lib64/httpd/modules/mod_wsgi-py26.so"
WSGIPythonHome "/usr"

PythonHome is not /usr, that is wrong.

**Comment by GrahamDumpleton** [comment]
The value:

```
WSGIPythonHome "/usr"
```

is correct, although for system Python installation as is being used here isn't required.

The question is, did you actually add that to your Apache configuration and have you added any other configuration to your Apache installation (if using system Apache and not mod_wsgi-express start-server), so it knows about your Flask application? What did you add?

**Comment by user7766666** [comment]
Sorry, seems like my copy pates did not work properly for flasktest.conf:

<VirtualHost *>
WSGIScriptAlias /flasktest /var/www/html/flasktest/flasktest.wsgi
WSGIPythonPath /usr/lib64/
ErrorLog /var/www/html/flasktest/error.log
<Directory /var/www/html/flasktest>
  WSGIProcessGroup flasktest
  WSGIApplicationGroup %{GLOBAL} 
  Order deny,allow
        Allow from all
</Directory>
</VirtualHost>

**Comment by user7766666** [comment]
I suppose it is removing parf of my code due to security reasons, but hopefully you understand the content of my 2 configuration files.

**Comment by user7766666** [comment]
Ok, some futher troubleshooting lead me to understand that httpd.conf needed to be updated.

LoadModule wsgi_module modules/mod_wsgi-py.so    <--- Pointing to 4.6.0 version

changed to

LoadModule wsgi_module modules/mod_wsgi-py26.so  <-- Pointing to 4.5.24 version, correct version.


When launching mydomain.com/flasktest I get the following in /var/log/httpd/error.log:

[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24274): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24277): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24278): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24279): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24278): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24280): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24279): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24280): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24281): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24281): Imported 'mod_wsgi'.

Seems like Apache managed to import mod_wsgi as I did not see the previosly, But the next question is what really happens as my Flask App is not launched.

**Comment by user7766666** [comment]
Looking further in /var/log/httpd/error.log, Apache seems to initialize several instances of mod_wsgi at the start and terminating it several times when shuting down Apache? Is behaviour correct? Wonder in case I interpret the logs correctly.


[Mon Mar 05 14:56:48 2018] [notice] suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
[Mon Mar 05 14:56:48 2018] [info] Init: Seeding PRNG with 256 bytes of entropy
[Mon Mar 05 14:56:48 2018] [info] Init: Initializing (virtual) servers for SSL
[Mon Mar 05 14:56:48 2018] [info] mod_ssl/2.2.15 compiled against Server: Apache/2.2.15, Library: OpenSSL/1.0.1e-fips
[Mon Mar 05 14:56:48 2018] [notice] Digest: generating secret for digest authentication ...
[Mon Mar 05 14:56:48 2018] [notice] Digest: done
[Mon Mar 05 14:56:48 2018] [debug] util_ldap.c(2089): LDAP merging Shared Cache conf: shm=0x7f7d6a289198 rmm=0x7f7d6a2891f0 for VHOST:
 seliicmon00005.seli.gic.ericsson.se
[Mon Mar 05 14:56:48 2018] [info] APR LDAP: Built with OpenLDAP LDAP SDK
[Mon Mar 05 14:56:48 2018] [info] LDAP: SSL support available
[Mon Mar 05 14:56:48 2018] [info] Init: Seeding PRNG with 256 bytes of entropy
[Mon Mar 05 14:56:48 2018] [debug] ssl_scache_shmcb.c(253): shmcb_init allocated 512000 bytes of shared memory
[Mon Mar 05 14:56:48 2018] [debug] ssl_scache_shmcb.c(272): for 511920 bytes (512000 including header), recommending 32 subcaches, 133
 indexes each
[Mon Mar 05 14:56:48 2018] [debug] ssl_scache_shmcb.c(306): shmcb_init_memory choices follow
[Mon Mar 05 14:56:48 2018] [debug] ssl_scache_shmcb.c(308): subcache_num = 32
[Mon Mar 05 14:56:48 2018] [debug] ssl_scache_shmcb.c(310): subcache_size = 15992
[Mon Mar 05 14:56:48 2018] [debug] ssl_scache_shmcb.c(312): subcache_data_offset = 3208
[Mon Mar 05 14:56:48 2018] [debug] ssl_scache_shmcb.c(314): subcache_data_size = 12784
[Mon Mar 05 14:56:48 2018] [debug] ssl_scache_shmcb.c(316): index_num = 133
[Mon Mar 05 14:56:48 2018] [info] Shared memory session cache initialised
[Mon Mar 05 14:56:48 2018] [info] Init: Initializing (virtual) servers for SSL
[Mon Mar 05 14:56:48 2018] [info] mod_ssl/2.2.15 compiled against Server: Apache/2.2.15, Library: OpenSSL/1.0.1e-fips
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 129801 for worker proxy:reverse
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 129801 for (*)
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 129802 for worker proxy:reverse
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 129802 for (*)
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 129803 for worker proxy:reverse
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 129803 for (*)
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 129804 for worker proxy:reverse
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 129804 for (*)
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 129805 for worker proxy:reverse
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 129805 for (*)
[Mon Mar 05 14:56:48 2018] [notice] Apache/2.2.15 (Unix) DAV/2 PHP/5.3.3 mod_ssl/2.2.15 OpenSSL/1.0.1e-fips configured -- resuming nor
mal operations
[Mon Mar 05 14:56:48 2018] [info] Server built: Sep 19 2017 11:18:50
[Mon Mar 05 14:56:48 2018] [debug] prefork.c(1028): AcceptMutex: sysvsem (default: sysvsem)
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 129806 for worker proxy:reverse
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 129806 for (*)
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 129807 for worker proxy:reverse
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 129807 for (*)
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 129808 for worker proxy:reverse
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Mon Mar 05 14:56:48 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 129808 for (*)
[Tue Mar 06 10:56:20 2018] [info] removed PID file /etc/httpd/run/httpd.pid (pid=129799)
[Tue Mar 06 10:56:20 2018] [notice] caught SIGTERM, shutting down
[Tue Mar 06 10:56:20 2018] [notice] suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
[Tue Mar 06 10:56:20 2018] [info] Init: Seeding PRNG with 256 bytes of entropy
[Tue Mar 06 10:56:20 2018] [info] Init: Initializing (virtual) servers for SSL
[Tue Mar 06 10:56:20 2018] [info] mod_ssl/2.2.15 compiled against Server: Apache/2.2.15, Library: OpenSSL/1.0.1e-fips
[Tue Mar 06 10:56:20 2018] [notice] Digest: generating secret for digest authentication ...
[Tue Mar 06 10:56:20 2018] [notice] Digest: done
[Tue Mar 06 10:56:20 2018] [debug] util_ldap.c(2089): LDAP merging Shared Cache conf: shm=0x7fefb10d8198 rmm=0x7fefb10d81f0 for VHOST:
 seliicmon00005.seli.gic.ericsson.se
[Tue Mar 06 10:56:20 2018] [info] APR LDAP: Built with OpenLDAP LDAP SDK
[Tue Mar 06 10:56:20 2018] [info] LDAP: SSL support available
[Tue Mar 06 10:56:20 2018] [info] Init: Seeding PRNG with 256 bytes of entropy
[Tue Mar 06 10:56:20 2018] [debug] ssl_scache_shmcb.c(253): shmcb_init allocated 512000 bytes of shared memory
[Tue Mar 06 10:56:20 2018] [debug] ssl_scache_shmcb.c(272): for 511920 bytes (512000 including header), recommending 32 subcaches, 133
 indexes each
[Tue Mar 06 10:56:20 2018] [debug] ssl_scache_shmcb.c(306): shmcb_init_memory choices follow
[Tue Mar 06 10:56:20 2018] [debug] ssl_scache_shmcb.c(308): subcache_num = 32
[Tue Mar 06 10:56:20 2018] [debug] ssl_scache_shmcb.c(310): subcache_size = 15992
[Tue Mar 06 10:56:20 2018] [debug] ssl_scache_shmcb.c(312): subcache_data_offset = 3208
[Tue Mar 06 10:56:20 2018] [debug] ssl_scache_shmcb.c(314): subcache_data_size = 12784
[Tue Mar 06 10:56:20 2018] [debug] ssl_scache_shmcb.c(316): index_num = 133
Tue Mar 06 10:56:20 2018] [info] Shared memory session cache initialised
[Tue Mar 06 10:56:20 2018] [info] Init: Initializing (virtual) servers for SSL
[Tue Mar 06 10:56:20 2018] [info] mod_ssl/2.2.15 compiled against Server: Apache/2.2.15, Library: OpenSSL/1.0.1e-fips
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 24274 for worker proxy:reverse
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 24274 for (*)
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24274): Initializing Python.
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 24275 for worker proxy:reverse
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 24275 for (*)
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24275): Initializing Python.
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 24276 for worker proxy:reverse
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 24276 for (*)
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24276): Initializing Python.
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 24277 for worker proxy:reverse
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 24277 for (*)
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24277): Initializing Python.
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 24278 for worker proxy:reverse
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 24278 for (*)
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24278): Initializing Python.
[Tue Mar 06 10:56:20 2018] [notice] Apache/2.2.15 (Unix) DAV/2 mod_wsgi/4.5.24 Python/2.6 PHP/5.3.3 mod_ssl/2.2.15 OpenSSL/1.0.1e-fips configured -- resuming normal operations
[Tue Mar 06 10:56:20 2018] [info] Server built: Sep 19 2017 11:18:50
[Tue Mar 06 10:56:20 2018] [debug] prefork.c(1028): AcceptMutex: sysvsem (default: sysvsem)
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 24279 for worker proxy:reverse
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 24279 for (*)
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24279): Initializing Python.
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 24280 for worker proxy:reverse
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 24280 for (*)
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24280): Initializing Python.
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1909): proxy: grabbed scoreboard slot 0 in child 24281 for worker proxy:reverse
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(1929): proxy: worker proxy:reverse already initialized
[Tue Mar 06 10:56:20 2018] [debug] proxy_util.c(2025): proxy: initialized single connection worker 0 in child 24281 for (*)
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24281): Initializing Python.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24274): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24275): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24276): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24277): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24276): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24275): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24274): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24277): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24278): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24279): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24278): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24280): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24279): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24280): Imported 'mod_wsgi'.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24281): Attach interpreter ''.
[Tue Mar 06 10:56:20 2018] [info] mod_wsgi (pid=24281): Imported 'mod_wsgi'.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24280): Destroying interpreters.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24280): Cleanup interpreter ''.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24281): Destroying interpreters.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24274): Destroying interpreters.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24275): Destroying interpreters.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24281): Cleanup interpreter ''.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24276): Destroying interpreters.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24274): Cleanup interpreter ''.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24275): Cleanup interpreter ''.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24276): Cleanup interpreter ''.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24278): Destroying interpreters.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24278): Cleanup interpreter ''.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24277): Destroying interpreters.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24279): Destroying interpreters.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24277): Cleanup interpreter ''.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24279): Cleanup interpreter ''.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24280): Terminating Python.
Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24281): Terminating Python.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24276): Terminating Python.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24278): Terminating Python.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24274): Terminating Python.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24275): Terminating Python.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24279): Terminating Python.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24277): Terminating Python.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24280): Python has shutdown.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24281): Python has shutdown.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24276): Python has shutdown.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24278): Python has shutdown.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24275): Python has shutdown.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24274): Python has shutdown.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24279): Python has shutdown.
[Tue Mar 06 11:10:47 2018] [info] mod_wsgi (pid=24277): Python has shutdown.
[Tue Mar 06 11:10:47 2018] [info] removed PID file /etc/httpd/run/httpd.pid (pid=24272)
[Tue Mar 06 11:10:47 2018] [notice] caught SIGTERM, shutting down
[

**Comment by GrahamDumpleton** [comment]
You have a few things not right. You have:

```
<VirtualHost *>
WSGIScriptAlias /flasktest /var/www/html/flasktest/flasktest.wsgi
WSGIPythonPath /usr/lib64/
ErrorLog /var/www/html/flasktest/error.log
<Directory /var/www/html/flasktest>
  WSGIProcessGroup flasktest
  WSGIApplicationGroup %{GLOBAL} 
  Order deny,allow
        Allow from all
</Directory>
</VirtualHost>
```

The first issue is that you are using:

```
<VirtualHost *>
```

You would usually qualify it with the port.

```
<VirtualHost *:80>
```

Second is that you are missing a ``ServerName`` directive inside of the ``VirtualHost``, which means that Apache isn't going to use it for name based virtual host matching. For that reason, if there is a default ``VirtualHost``, or other ``VirtualHost`` that comes before it, it will use that instead.

Suggest adding:

```
ServerName 127.0.0.1.nip.io
```

inside of the ``VirtualHost`` and use as the URL ``http://127.0.0.1.nip.io/flasktest``.

A final issue is that you are putting your code under ``/var/www/html``. That is usually a bad idea as that is the Apache ``DocumentRoot`` directory for some Linux distros. This means someone could potentially download code you have in that directory. Put your code in a different directory.

**Comment by user7766666** [comment]
Made the changes you proposed, except chaning directory from '/var/www/html/'. Unfortunately a decision was made in the past to use that diretory for our applications which I can't change (this Apache server is only used for internal development so it should be fine)

Still fighting to get my code to run.  There is no change in behaviour. Is there a way to enable more logging to understand where things get stuck?

**Comment by GrahamDumpleton** [comment]
Where do you have:

```
WSGIDaemonProcess flasktest
```

defined?

If you don't have it, add it inside of the ``VirtualHost``.

For good measure, add outside of the ``VirtualHost``:

```
WSGIRestrictEmbedded On
```

If ``WSGIDaemonProcess`` was missing, you should have seen an error logged.

What does your configuration look like now and what URL are you using? What error do you get in the browser?

**Comment by user7766666** [comment]
I have modified my flasktest.conf as you suggested:

...
<VirtualHost *:80>
ServerName xxxxx.xxxxxx.xxxxxx.xx
WSGIScriptAlias /flasktest /var/www/html/flasktest/flasktest.wsgi
WSGIPythonPath /usr/lib64/
WSGIDaemonProcess flasktest
ErrorLog /var/www/html/flasktest/error.log
<Directory /var/www/html/flasktest>
  WSGIProcessGroup flasktest
  WSGIApplicationGroup %{GLOBAL} 
  Order deny,allow
   Allow from all
</Directory>
</VirtualHost>
WSGIRestrictEmbedded On
...

I'm accessing  xxxxx.xxxxxx.xxxxxx.xx/flasktest and the only thing that happens is that I get a directory listing showing the files within /var/www/html/flasktest. No error message at all.

**Comment by user7766666** [comment]
```
<VirtualHost *:80>
ServerName  xxxxx.xxxxxx.xxxxxx.xx
WSGIScriptAlias /flasktest /var/www/html/flasktest/flasktest.wsgi
WSGIPythonPath /usr/lib64/
WSGIDaemonProcess flasktest
ErrorLog /var/www/html/flasktest/error.log
<Directory /var/www/html/flasktest>
  WSGIProcessGroup flasktest
  WSGIApplicationGroup %{GLOBAL} 
  Order deny,allow
        Allow from all
</Directory>
</VirtualHost>
WSGIRestrictEmbedded On
```

**Comment by GrahamDumpleton** [comment]
Did you do what I told you earlier, which was to use:

```
ServerName 127.0.0.1.nip.io
```

inside of the VirtualHost and use as the URL ``http://127.0.0.1.nip.io/flasktest``.

You can't use an IP address if you still have a default ``VirtualHost`` as it will ignore this new one and use it, because an IP address in URL will not match name based virtual hosting configuration. This is why I told you to use ``127.0.0.1.nip.io``. Otherwise you need to disable the default ``VirtualHost`` so that this new one is the only one enabled, because when it can't match by name, Apache will fall back to using the first ``VirtualHost`` it found.

**Comment by GrahamDumpleton** [comment]
The original issue with ``PyFrame_GetLineNumber`` being undefined was fixed in 4.6.3.
